discovery/discovery_test.go

```diff
@@ -335,26 +267,104 @@ func TestNewDBOrchestratorPoolCache_TestURLs_Empty(t *testing.T) {
 	assert := assert.New(t)
 	require.Nil(err)
 
-	node, _ := core.NewLivepeerNode(nil, "", nil)
-	node.Database = dbh
-
 	addresses := []string{}
 	// Addresses is empty slice -> No orchestrators
 	orchestrators := StubOrchestrators(addresses)
-	// Contains empty orchestrator slice -> No registered orchs found on chain error
-	node.Eth = &eth.StubClient{Orchestrators: orchestrators}
-	errorLogsBefore := glog.Stats.Error.Lines()
-	dbOrch := NewDBOrchestratorPoolCache(node)
-	errorLogsAfter := glog.Stats.Error.Lines()
+	for _, o := range orchestrators {
+		dbh.UpdateOrch(ethOrchToDBOrch(o))
+	}
+	dbOrch := NewDBOrchestratorPoolCache(dbh, &eth.StubClient{}, &stubRoundsManager{}, &stubTicketParamsValidator{})
+	require.NotNil(dbOrch)
 	require.NotNil(dbOrch)
 	assert.Equal(0, dbOrch.Size())
-	assert.NotZero(t, errorLogsAfter-errorLogsBefore)
 	urls := dbOrch.GetURLs()
 	assert.Len(urls, 0)
 }
 
+func TestNewDBOrchestorPoolCache_PollOrchestratorInfo(t *testing.T) {
+	cachedOrchInfo := &net.OrchestratorInfo{
+		Transcoder: "transcoderFromTest",
+		PriceInfo: &net.PriceInfo{
+			PricePerUnit:  999,
+			PixelsPerUnit: 1,
+		},
+	}
+	polledOrchInfo := &net.OrchestratorInfo{
+		Transcoder: "transcoderFromTest",
+		PriceInfo: &net.PriceInfo{
+			PricePerUnit:  1,
+			PixelsPerUnit: 1,
+		},
+	}
+	returnInfo := cachedOrchInfo
+
+	var mu sync.Mutex
+	first := true
+	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
+		mu.Lock()
+		if first {
+			time.Sleep(100 * time.Millisecond)
+			first = false
+		}
+		mu.Unlock()
+		return returnInfo, nil
+	}
+
+	dbh, dbraw, err := common.TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	require := require.New(t)
+	assert := assert.New(t)
+	require.Nil(err)
+
+	// adding orchestrators to DB
+	addresses := []string{"https://127.0.0.1:8936", "https://127.0.0.1:8937", "https://127.0.0.1:8938"}
+	orchestrators := StubOrchestrators(addresses)
+
+	testOrchs := make([]orchTest, 0)
+	expPrice, _ := common.PriceToFixed(big.NewRat(999, 1))
+	for _, o := range orchestrators {
+		to := orchTest{
+			EthereumAddr:  o.Address.String(),
+			ServiceURI:    o.ServiceURI,
+			PricePerPixel: expPrice,
+		}
+		testOrchs = append(testOrchs, to)
+	}
+
+	ethClient := &eth.StubClient{Orchestrators: orchestrators}
+
+	pool := NewDBOrchestratorPoolCache(dbh, ethClient, &stubRoundsManager{}, &stubTicketParamsValidator{})
+
+	// Add orchestrators to DB
+	err = pool.CacheTranscoderPool(nil)
+	require.Nil(err)
+
+	// Ensure orchestrators exist in DB
+	require.Equal(pool.Size(), 3)
+	dbOrchs, err := pool.store.SelectOrchs(nil)
+	require.Nil(err)
+	require.Len(dbOrchs, 3)
+	for _, o := range dbOrchs {
+		test := toOrchTest(o.EthereumAddr, o.ServiceURI, o.PricePerPixel)
+		require.Contains(testOrchs, test)
+	}
+
+	cacheRefreshInterval = 1 * time.Second
+	returnInfo = polledOrchInfo
+	pool.PollOrchestratorInfo(nil)
```

Comment:
Something that could be nice to test is that based on the `cacheRefreshInterval` value and the sleep time that we actually requested info for a URI a certain # of times.

Reply:
Good idea, would the new test in 9458ef19 suffice?